### PR TITLE
[audio_video_recorder][jsk_rosbag_tools] support ros-o

### PR DESCRIPTION
**Pull request by mqcmd196**

### CODE REVIEW

**Comment by k-okada** [comment]
@mqcmd196 could you check if `test_bag_to_video` works on your ROS-O environment?

```
  <?xml version="1.0" encoding="utf-8"?>
  <testsuite errors="0" failures="1" name="unittest.suite.TestSuite" tests="1" time="0.906"><testcase classname="rostest.runner.RosTest" name="testtest_bag_to_video" time="0.9062"><failure type="AssertionError">False is not true : test [test_bag_to_video] did not generate test results
    File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
      yield
    File "/usr/lib/python3.8/unittest/case.py", line 676, in run
      self._callTestMethod(testMethod)
    File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
      method()
    File "/opt/ros/noetic/lib/python3/dist-packages/rostest/runner.py", line 164, in fn
      self.assert_(os.path.isfile(test_file), "test [%s] did not generate test results"%test_name)
    File "/usr/lib/python3.8/unittest/case.py", line 1410, in deprecated_func
      return original_func(*args, **kwargs)
    File "/usr/lib/python3.8/unittest/case.py", line 765, in assertTrue
      raise self.failureException(msg)
  </failure></testcase><system-out>&lt;![CDATA[
```

File: jsk_rosbag_tools/CMakeLists.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@ catkin_package(
   CATKIN_DEPENDS
 )
 
-if($ENV{ROS_DISTRO} STREQUAL "noetic")
+if($ENV{ROS_DISTRO} STRGREATER "noetic")
```
Comment (k-okada) [inline jsk_rosbag_tools/CMakeLists.txt]: `STRGREATER` does not include "noetic", need to use `STRGREATER_EQUAL`

**Comment by mqcmd196** [comment]
@k-okada 
Seems working with 88f6c1f5

```
❯ catkin test --this
Starting  >>> jsk_rosbag_tools
_________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________________
Warnings   << jsk_rosbag_tools:make /home/obinata/ros/catkin_ws/logs/jsk_rosbag_tools/test.make.003.log

[notice] A new release of pip available: 22.2.2 -> 24.3.1
[notice] To update, run: /home/obinata/ros/catkin_ws/build/jsk_rosbag_tools/venv/bin/python -m pip install --upgrade pip

[notice] A new release of pip available: 22.2.2 -> 24.3.1
[notice] To update, run: /home/obinata/ros/catkin_ws/build/jsk_rosbag_tools/venv/bin/python -m pip install --upgrade pip
-- run_tests.py: execute commands
  /usr/bin/python3 /opt/ros/one/share/rostest/cmake/../../../bin/rostest --pkgdir=/home/obinata/ros/catkin_ws/src/jsk-ros-pkg/jsk_common/jsk_rosbag_tools --package=jsk_rosbag_tools --results-filename tests_test_bag_to_video.xml --results-base-dir "/home/obinata/ros/catkin_ws/build/jsk_rosbag_tools/test_results" /home/obinata/ros/catkin_ws/src/jsk-ros-pkg/jsk_common/jsk_rosbag_tools/tests/test_bag_to_video.test
... logging to /home/obinata/.ros/log/rostest-sinobuz-16846.log
[ROSUNIT] Outputting test results to /home/obinata/ros/catkin_ws/build/jsk_rosbag_tools/test_results/jsk_rosbag_tools/rostest-tests_test_bag_to_video.xml

  0% 0/173 [00:00<?, ?it/s]
  2% 4/173 [00:00<00:04, 37.04it/s]
  6% 11/173 [00:00<00:03, 53.77it/s]
 10% 17/173 [00:00<00:02, 55.99it/s]
 14% 24/173 [00:00<00:02, 59.01it/s]
 18% 31/173 [00:00<00:02, 60.59it/s]
 22% 38/173 [00:00<00:02, 58.34it/s]
 25% 44/173 [00:00<00:02, 58.55it/s]
 29% 50/173 [00:00<00:02, 58.22it/s]
 33% 57/173 [00:00<00:01, 60.58it/s]
 37% 64/173 [00:01<00:01, 61.14it/s]
 41% 71/173 [00:01<00:01, 60.84it/s]
 45% 78/173 [00:01<00:01, 61.12it/s]
 49% 85/173 [00:01<00:01, 61.15it/s]
 53% 92/173 [00:01<00:01, 60.30it/s]
 57% 99/173 [00:01<00:01, 58.53it/s]
 61% 105/173 [00:01<00:01, 58.73it/s]
 64% 111/173 [00:01<00:01, 57.41it/s]
 68% 117/173 [00:02<00:01, 55.68it/s]
 71% 123/173 [00:02<00:00, 56.22it/s]
 75% 130/173 [00:02<00:00, 59.65it/s]
 79% 137/173 [00:02<00:00, 61.57it/s]
 83% 144/173 [00:02<00:00, 62.35it/s]
 87% 151/173 [00:02<00:00, 63.94it/s]
 91% 158/173 [00:02<00:00, 65.04it/s]
 95% 165/173 [00:02<00:00, 65.54it/s]
 99% 172/173 [00:02<00:00, 66.58it/s]
100% 173/173 [00:02<00:00, 58.93it/s]

chunk:   0% 0/883 [00:00<?, ?it/s, now=None]
chunk:  32% 280/883 [00:00<00:00, 2799.56it/s, now=None]
chunk:  67% 595/883 [00:00<00:00, 2996.68it/s, now=None]


t:   0% 0/1197 [00:00<?, ?it/s, now=None]
t:   5% 62/1197 [00:00<00:01, 618.75it/s, now=None]
t:  10% 124/1197 [00:00<00:02, 506.36it/s, now=None]
t:  15% 176/1197 [00:00<00:02, 507.46it/s, now=None]
t:  19% 228/1197 [00:00<00:01, 489.99it/s, now=None]
t:  23% 278/1197 [00:00<00:01, 485.69it/s, now=None]
t:  27% 327/1197 [00:00<00:01, 469.40it/s, now=None]
t:  31% 375/1197 [00:00<00:01, 471.52it/s, now=None]
t:  35% 423/1197 [00:00<00:01, 469.92it/s, now=None]
t:  39% 471/1197 [00:00<00:01, 457.89it/s, now=None]
t:  43% 517/1197 [00:01<00:01, 441.66it/s, now=None]
t:  47% 568/1197 [00:01<00:01, 460.42it/s, now=None]
t:  51% 615/1197 [00:01<00:01, 435.18it/s, now=None]
t:  55% 661/1197 [00:01<00:01, 421.22it/s, now=None]
t:  59% 712/1197 [00:01<00:01, 436.23it/s, now=None]
t:  63% 758/1197 [00:01<00:00, 440.96it/s, now=None]
t:  67% 807/1197 [00:01<00:00, 452.00it/s, now=None]
t:  72% 857/1197 [00:01<00:00, 463.92it/s, now=None]
t:  76% 905/1197 [00:01<00:00, 465.98it/s, now=None]
t:  80% 962/1197 [00:02<00:00, 493.94it/s, now=None]
t:  85% 1015/1197 [00:02<00:00, 502.81it/s, now=None]
t:  89% 1067/1197 [00:02<00:00, 505.50it/s, now=None]
t:  93% 1119/1197 [00:02<00:00, 509.44it/s, now=None]
t:  98% 1171/1197 [00:02<00:00, 512.28it/s, now=None]


  0% 0/173 [00:00<?, ?it/s]
  9% 16/173 [00:00<00:01, 156.93it/s]
 43% 74/173 [00:00<00:00, 401.11it/s]
 80% 139/173 [00:00<00:00, 512.56it/s]
100% 173/173 [00:00<00:00, 280.69it/s]

chunk:   0% 0/883 [00:00<?, ?it/s, now=None]
chunk:  29% 255/883 [00:00<00:00, 2518.16it/s, now=None]
chunk:  64% 567/883 [00:00<00:00, 2870.17it/s, now=None]
chunk:  99% 878/883 [00:00<00:00, 2976.36it/s, now=None]


t:   0% 0/40 [00:00<?, ?it/s, now=None]


  0% 0/173 [00:00<?, ?it/s]
  2% 3/173 [00:00<00:06, 25.56it/s]
  4% 7/173 [00:00<00:05, 32.03it/s]
  6% 11/173 [00:00<00:04, 33.87it/s]
  9% 15/173 [00:00<00:04, 34.59it/s]
 11% 19/173 [00:00<00:04, 35.04it/s]
 13% 23/173 [00:00<00:04, 33.84it/s]
 16% 27/173 [00:00<00:04, 34.95it/s]
 18% 31/173 [00:00<00:04, 35.37it/s]
 20% 35/173 [00:01<00:04, 33.55it/s]
 23% 39/173 [00:01<00:04, 31.83it/s]
 25% 43/173 [00:01<00:03, 32.53it/s]
 27% 47/173 [00:01<00:03, 31.75it/s]
 29% 51/173 [00:01<00:03, 33.45it/s]
 32% 55/173 [00:01<00:03, 33.14it/s]
 34% 59/173 [00:01<00:03, 32.91it/s]
 36% 63/173 [00:01<00:03, 34.54it/s]
 39% 67/173 [00:01<00:03, 34.52it/s]
 41% 71/173 [00:02<00:02, 34.87it/s]
 43% 75/173 [00:02<00:02, 35.49it/s]
 46% 79/173 [00:02<00:02, 35.00it/s]
 48% 83/173 [00:02<00:02, 34.98it/s]
 50% 87/173 [00:02<00:02, 34.56it/s]
 53% 91/173 [00:02<00:02, 34.18it/s]
 55% 95/173 [00:02<00:02, 34.82it/s]
 57% 99/173 [00:02<00:02, 33.02it/s]
 60% 103/173 [00:03<00:02, 31.60it/s]
 62% 107/173 [00:03<00:02, 32.45it/s]
 64% 111/173 [00:03<00:01, 33.41it/s]
 66% 115/173 [00:03<00:01, 32.77it/s]
 69% 119/173 [00:03<00:01, 33.58it/s]
 71% 123/173 [00:03<00:01, 35.02it/s]
 73% 127/173 [00:03<00:01, 35.08it/s]
 76% 131/173 [00:03<00:01, 35.86it/s]
 78% 135/173 [00:03<00:01, 35.67it/s]
 80% 139/173 [00:04<00:00, 36.60it/s]
 83% 143/173 [00:04<00:00, 36.58it/s]
 85% 147/173 [00:04<00:00, 36.79it/s]
 87% 151/173 [00:04<00:00, 35.67it/s]
 90% 155/173 [00:04<00:00, 36.34it/s]
 92% 159/173 [00:04<00:00, 36.00it/s]
 94% 163/173 [00:04<00:00, 36.65it/s]
 97% 167/173 [00:04<00:00, 36.82it/s]
 99% 171/173 [00:04<00:00, 36.40it/s]
100% 173/173 [00:05<00:00, 34.05it/s]
[Testcase: testtest_bag_to_video] ... ok

[ROSTEST]-----------------------------------------------------------------------

[jsk_rosbag_tools.rosunit-test_bag_to_video/test_bag_to_video_and_video_to_bag_and_compress][passed]
[jsk_rosbag_tools.rosunit-test_bag_to_video/test_bag_to_video_fps][passed]

SUMMARY
 * RESULT: SUCCESS
 * TESTS: 2
 * ERRORS: 0
 * FAILURES: 0

rostest log file is in /home/obinata/.ros/log/rostest-sinobuz-16846.log
-- run_tests.py: execute commands
  /usr/bin/python3 /opt/ros/one/share/rostest/cmake/../../../bin/rostest --pkgdir=/home/obinata/ros/catkin_ws/src/jsk-ros-pkg/jsk_common/jsk_rosbag_tools --package=jsk_rosbag_tools --results-filename tests_test_jsk_rosbag_tools.xml --results-base-dir "/home/obinata/ros/catkin_ws/build/jsk_rosbag_tools/test_results" /home/obinata/ros/catkin_ws/src/jsk-ros-pkg/jsk_common/jsk_rosbag_tools/tests/test_jsk_rosbag_tools.test
... logging to /home/obinata/.ros/log/rostest-sinobuz-17845.log
[ROSUNIT] Outputting test results to /home/obinata/ros/catkin_ws/build/jsk_rosbag_tools/test_results/jsk_rosbag_tools/rostest-tests_test_jsk_rosbag_tools.xml
[Testcase: testtest_jsk_rosbag_tools] ... ok

[ROSTEST]-----------------------------------------------------------------------

[jsk_rosbag_tools.rosunit-test_jsk_rosbag_tools/test_bag_to_audio][passed]
[jsk_rosbag_tools.rosunit-test_jsk_rosbag_tools/test_merge][passed]
[jsk_rosbag_tools.rosunit-test_jsk_rosbag_tools/test_tf_static_to_tf][passed]

SUMMARY
 * RESULT: SUCCESS
 * TESTS: 3
 * ERRORS: 0
 * FAILURES: 0

rostest log file is in /home/obinata/.ros/log/rostest-sinobuz-17845.log
-- run_tests.py: execute commands with working directory "/home/obinata/ros/catkin_ws/build/jsk_rosbag_tools"
  /home/obinata/ros/catkin_ws/build/jsk_rosbag_tools/catkin_generated/env_cached.sh rosrun catkin_virtualenv venv_check venv --requirements /home/obinata/ros/catkin_ws/src/jsk-ros-pkg/jsk_common/jsk_rosbag_tools/requirements.txt         --extra-pip-args "\"-qq --retries 10 --timeout 30\""         --xunit-output /home/obinata/ros/catkin_ws/build/jsk_rosbag_tools/test_results/jsk_rosbag_tools/venv_check-jsk_rosbag_tools-requirements.xml
-- run_tests.py: execute commands with working directory "/home/obinata/ros/catkin_ws/build/jsk_rosbag_tools"
  /opt/ros/one/share/roslint/cmake/../../../lib/roslint/test_wrapper /home/obinata/ros/catkin_ws/build/jsk_rosbag_tools/test_results/jsk_rosbag_tools/roslint-jsk_rosbag_tools.xml make roslint_jsk_rosbag_tools

chunk:   0% 0/883 [00:00<?, ?it/s, now=None]
chunk:  31% 276/883 [00:00<00:00, 2755.21it/s, now=None]
chunk:  66% 581/883 [00:00<00:00, 2923.32it/s, now=None]


t:   0% 0/2391 [00:00<?, ?it/s, now=None]
t:   3% 61/2391 [00:00<00:03, 606.67it/s, now=None]
t:   5% 122/2391 [00:00<00:04, 515.14it/s, now=None]
t:   7% 175/2391 [00:00<00:04, 502.85it/s, now=None]
t:  10% 233/2391 [00:00<00:04, 527.90it/s, now=None]
t:  12% 287/2391 [00:00<00:04, 525.62it/s, now=None]
t:  14% 340/2391 [00:00<00:03, 526.23it/s, now=None]
t:  17% 396/2391 [00:00<00:03, 535.30it/s, now=None]
t:  19% 453/2391 [00:00<00:03, 538.87it/s, now=None]
t:  21% 507/2391 [00:00<00:03, 533.10it/s, now=None]
t:  23% 561/2391 [00:01<00:03, 535.17it/s, now=None]
t:  26% 615/2391 [00:01<00:03, 517.83it/s, now=None]
t:  28% 671/2391 [00:01<00:03, 530.15it/s, now=None]
t:  30% 727/2391 [00:01<00:03, 536.21it/s, now=None]
t:  33% 781/2391 [00:01<00:03, 503.99it/s, now=None]
t:  35% 839/2391 [00:01<00:02, 524.63it/s, now=None]
t:  37% 896/2391 [00:01<00:02, 534.17it/s, now=None]
t:  40% 950/2391 [00:01<00:02, 524.17it/s, now=None]
t:  42% 1004/2391 [00:01<00:02, 527.09it/s, now=None]
t:  44% 1062/2391 [00:02<00:02, 540.71it/s, now=None]
t:  47% 1117/2391 [00:02<00:02, 538.46it/s, now=None]
t:  49% 1171/2391 [00:02<00:02, 535.36it/s, now=None]
t:  51% 1225/2391 [00:02<00:02, 524.12it/s, now=None]
t:  53% 1278/2391 [00:02<00:02, 524.19it/s, now=None]
t:  56% 1331/2391 [00:02<00:02, 512.94it/s, now=None]
t:  58% 1383/2391 [00:02<00:02, 494.19it/s, now=None]
t:  60% 1441/2391 [00:02<00:01, 516.06it/s, now=None]
t:  63% 1499/2391 [00:02<00:01, 532.67it/s, now=None]
t:  65% 1558/2391 [00:02<00:01, 547.40it/s, now=None]
t:  68% 1614/2391 [00:03<00:01, 550.15it/s, now=None]
t:  70% 1672/2391 [00:03<00:01, 558.85it/s, now=None]
t:  72% 1728/2391 [00:03<00:01, 544.37it/s, now=None]
t:  75% 1783/2391 [00:03<00:01, 530.02it/s, now=None]
t:  77% 1837/2391 [00:03<00:01, 531.83it/s, now=None]
t:  79% 1892/2391 [00:03<00:00, 534.45it/s, now=None]
t:  81% 1946/2391 [00:03<00:00, 535.61it/s, now=None]
t:  84% 2000/2391 [00:03<00:00, 536.54it/s, now=None]
t:  86% 2055/2391 [00:03<00:00, 539.35it/s, now=None]
t:  88% 2111/2391 [00:03<00:00, 543.84it/s, now=None]
t:  91% 2168/2391 [00:04<00:00, 549.98it/s, now=None]
t:  93% 2224/2391 [00:04<00:00, 550.50it/s, now=None]
t:  95% 2280/2391 [00:04<00:00, 551.63it/s, now=None]
t:  98% 2336/2391 [00:04<00:00, 550.84it/s, now=None]

Built target roslint_jsk_rosbag_tools
cd /home/obinata/ros/catkin_ws/build/jsk_rosbag_tools; catkin test --get-env jsk_rosbag_tools | catkin env -si  /usr/bin/make run_tests; cd -

.........................................................................................................................................................................................................................................................................................................................................................................................
Output     << jsk_rosbag_tools:results /home/obinata/ros/catkin_ws/logs/jsk_rosbag_tools/test.results.003.log
Summary: 9 tests, 0 errors, 0 failures, 0 skipped
cd /home/obinata/ros/catkin_ws/build/jsk_rosbag_tools; catkin test --get-env jsk_rosbag_tools | catkin env -si  catkin_test_results; cd -

Finished  <<< jsk_rosbag_tools          [ 1 minute and 1.4 seconds ]
[test] Summary: All 1 packages succeeded!
[test]   Ignored:   None.
[test]   Warnings:  1 packages succeeded with warnings.
[test]   Abandoned: None.
[test]   Failed:    None.
[test] Runtime: 1 minute and 1.6 seconds total.
```

File: jsk_rosbag_tools/CMakeLists.txt (hunk 1 above)
Comment (mqcmd196) [reply jsk_rosbag_tools/CMakeLists.txt]: I think only using `STREQUAL` is better as [the advice](https://github.com/jsk-ros-pkg/jsk_3rdparty/issues/509#issuecomment-2482742076) . It is unpredictable what `ROS_DISTRO` will be in the future release. In my environment, the `ROS_DISTRO` is `one`

**Comment by k-okada** [comment]
@mqcmd196 can we delete `ros-o` branch?

**Comment by mqcmd196** [comment]
Yes, ok

**Pull request merged**